src/main/java/com/trillia/entities/Department.java

```diff
@@ -0,0 +1,121 @@
+package com.trillia.entities;
+
+import org.hibernate.search.annotations.*;
+import org.hibernate.search.annotations.Index;
+
+import javax.persistence.*;
+import javax.xml.bind.annotation.XmlElement;
+import java.io.Serializable;
+
+/**
+ * Created by mmathew on 27/05/16.
+ */
+
+
+@Entity
+@Table(name = "trl_departments", uniqueConstraints = {@UniqueConstraint(columnNames = {"tenantId"}), @UniqueConstraint(columnNames = {"departmentCode"})})
+@Indexed(index = "Department")
```

Comment:
Also, please use the @FullTextFilterDef(name="filter-dept-by-tenant", impl=TenantBasedSearchFilterFactory.class) so that the lucene queries are also automatically filtered by tenantId .   Just change the "name" for each entity.